auto_round/utils.py

```diff
@@ -366,19 +365,20 @@ def sampling_inputs(input_ids, input_others, indices, seqlen,
     current_input_ids = torch.cat(current_input_ids, dim=input_dim)
     current_input_others = {"positional_inputs": input_others["positional_inputs"]}
     for key in input_others.keys():
-        if not share_attention_mask_flag and ("attention_mask" in key or "alibi" in key) \
-                or (not_share_position_ids_flag and ("position_ids" in key or \
-                    "cache_position" in key or "position_embeddings" in key)) \
-                or (not_share_rotary_pos_emb_flag and ("rotary_pos_emb" in key or 'cu_seqlens' in key)) \
-                or "cross_attention_states" in key:
+        if "positional_inputs" in key:
+            continue
+        if (key not in shareable_keywords or len(indices) == 1) \
+                and not isinstance(input_others[key], (str, bool, type(None))):
             current_input_others[key] = None
             if input_others[key] is not None:
                 current_input_others[key] = [input_others[key][i] for i in indices]
-                if not isinstance(current_input_others[key], torch.Tensor):
-                    if len(current_input_others[key]) == 1:
-                        current_input_others[key] = current_input_others[key][0]
-                    else:
+                if len(indices) == 1:
+                    current_input_others[key] = current_input_others[key][0]
```

Comment:
should check the code is ok for the the coming transformers with kv cache class,